pkg/composableschemadsl/compiler/translator.go

```diff
@@ -30,7 +30,7 @@ type translationContext struct {
 	enabledFlags     []string
 	existingNames    *mapz.Set[string]
 	// The mapping of partial name -> relations represented by the partial
-	compiledPartials map[string][]*core.Relation
+	compiledPartials *map[string][]*core.Relation
```

Comment:
Is there a reason to make this a pointer-to-map? It looks like the other lines would be fine with just a regular map